File: test/language/testdata/python/ParameterTypes.py
from datetime import datetime

from cucumber_expressions.parameter_type import ParameterType

ParameterType(
    name="uuid",
    regexp="/[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}/",
    transformer=lambda x: x,
)

ParameterType(
    name="date",
    transformer=lambda x: datetime.datetime(x),
    regexp="/\d{4}-\d{2}-\d{2}/",
)

ParameterType(
    transformer=lambda x: x,
    regexp="jupiter|mars|tellus",
    name="planet",
)

File: test/language/testdata/ruby/ParameterTypes.rb
require 'date'

ParameterType(
  name:        'uuid',
  regexp:      /[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}/,
  transformer: ->(s) { s }
)

ParameterType(
  name:        'date',
  regexp:      /\d{4}-\d{2}-\d{2}/,
  transformer: ->(s) { Date.parse(s) }
)

ParameterType(
  name:        'planet',
  regexp:      ['jupiter', 'mars', 'tellus'],
  transformer: ->(s) { s }
)
